### PR TITLE
Nick rec index table

### LINKED ISSUE #15: Replace RecommendationTable placeholder page with one that includes Recommendation Table
- [x] Index Page displays Recommendation table columns
- [x] test coverage for new index page additions
- [x] create the Recommendation table component
- [x] add testing for Recommendation table

### PR DESCRIPTION
**Pull request by nick-pellegrin**
closes #15

### CODE REVIEW

**Review by josiahnross — APPROVED** [review]
LGTM!

File: frontend/src/main/pages/Recommendations/RecommendationsIndexPage.js [hunk 1]
```diff
@@ -1,13 +1,28 @@
...
 
-    return (
+  const currentUser = useCurrentUser();
+
+  const { data: recommendations, error: _error, status: _status } =
+    useBackend(
+      // Stryker disable next-line all : don't test internal caching of React Query
+      ["/api/recommendations/all"],
+            // Stryker disable next-line StringLiteral,ObjectLiteral : since "GET" is default, "" is an equivalent mutation
+            { method: "GET", url: "/api/recommendations/all" },
```
Comment (Gopu2001) [inline frontend/src/main/pages/Recommendations/RecommendationsIndexPage.js:17]: You should change this URL. When I test your branch, I get some errors in API requests. According to your Recommendation controller java file, the endpoint is defined as: `/api/Recommendation/all`
```suggestion
            { method: "GET", url: "/api/Recommendation/all" },
```

File: frontend/src/main/pages/Recommendations/RecommendationsIndexPage.js [hunk 2]
```diff
@@ -1,13 +1,28 @@
...
 
 export default function RecommendationsIndexPage() {
 
-    return (
+  const currentUser = useCurrentUser();
+
+  const { data: recommendations, error: _error, status: _status } =
+    useBackend(
+      // Stryker disable next-line all : don't test internal caching of React Query
+      ["/api/recommendations/all"],
```
Comment (Gopu2001) [inline frontend/src/main/pages/Recommendations/RecommendationsIndexPage.js:15]: Incorrect endpoint:
```suggestion
      ["/api/Recommendation/all"],
```

File: frontend/src/tests/components/Recommendations/RecommendationsTable.test.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    expect(getByTestId(`${testId}-cell-row-1-col-id`)).toHaveTextContent("2");
+    
+
+    // const editButton = getByTestId(`${testId}-cell-row-0-col-Edit-button`);
+    // expect(editButton).toBeInTheDocument();
+    // expect(editButton).toHaveClass("btn-primary");
+
+    // const deleteButton = getByTestId(`${testId}-cell-row-0-col-Delete-button`);
+    // expect(deleteButton).toBeInTheDocument();
+    // expect(deleteButton).toHaveClass("btn-danger");
```
Comment (Gopu2001) [inline frontend/src/tests/components/Recommendations/RecommendationsTable.test.js:96]: Nitpick: commented code should only be for documentation purposes. Though I understand if you want to keep this in for the later PR.

File: frontend/src/tests/pages/Recommendations/RecommendationsIndexPage.test.js [hunk 4]
```diff
@@ -78,6 +80,70 @@ describe("RecommendationsIndexPage tests", () => {
 
 
     });
+
+    test("renders three recommendations without crashing for regular user", async () => {
+        setupUserOnly();
+        const queryClient = new QueryClient();
+        axiosMock.onGet("/api/recommendations/all").reply(200, recommendationsFixtures.threeRecommendations);
```
Comment (Gopu2001) [inline frontend/src/tests/pages/Recommendations/RecommendationsIndexPage.test.js:87]: bad endpoint:
```suggestion
        axiosMock.onGet("/api/Recommendation/all").reply(200, recommendationsFixtures.threeRecommendations);
```

File: frontend/src/tests/pages/Recommendations/RecommendationsIndexPage.test.js [hunk 5]
```diff
@@ -78,6 +80,70 @@ describe("RecommendationsIndexPage tests", () => {
...
+        await waitFor(  () => { expect(getByTestId(`${testId}-cell-row-0-col-id`)).toHaveTextContent("1"); } );
+        expect(getByTestId(`${testId}-cell-row-1-col-id`)).toHaveTextContent("2");
+        expect(getByTestId(`${testId}-cell-row-2-col-id`)).toHaveTextContent("3");
+
+    });
+
+    test("renders three recommendations without crashing for admin user", async () => {
+        setupAdminUser();
+        const queryClient = new QueryClient();
+        axiosMock.onGet("/api/recommendations/all").reply(200, recommendationsFixtures.threeRecommendations);
```
Comment (Gopu2001) [inline frontend/src/tests/pages/Recommendations/RecommendationsIndexPage.test.js:106]: bad endpoint:
```suggestion
        axiosMock.onGet("/api/Recommendation/all").reply(200, recommendationsFixtures.threeRecommendations);
```

File: frontend/src/tests/pages/Recommendations/RecommendationsIndexPage.test.js [hunk 6]
```diff
@@ -78,6 +80,70 @@ describe("RecommendationsIndexPage tests", () => {
...
+        expect(getByTestId(`${testId}-cell-row-1-col-id`)).toHaveTextContent("2");
+        expect(getByTestId(`${testId}-cell-row-2-col-id`)).toHaveTextContent("3");
+
+    });
+
+    test("renders empty table when backend unavailable, user only", async () => {
+        setupUserOnly();
+
+        const queryClient = new QueryClient();
+        axiosMock.onGet("/api/recommendations/all").timeout();
```
Comment (Gopu2001) [inline frontend/src/tests/pages/Recommendations/RecommendationsIndexPage.test.js:126]: bad endpoint:
```suggestion
        axiosMock.onGet("/api/Recommendation/all").timeout();
```

**Review by Gopu2001 — CHANGES_REQUESTED** [review]
Hey Nick, overall good progress. One nitpick, otherwise there are some issues with the endpoint of the API you are calling. May you please fix them? Thanks!

**Review by Gopu2001 — APPROVED** [review]
Nice, everything works now!

**Pull request merged**